### ISSUE TITLE
Generated documentation includes bullet points that are not part of the docs

### ISSUE THREAD

**Issue by Skeletonxf** [rust-lang/rust#93309]
Hi, I've just published a new version of my crate and the generated documentation is including a lot of bullet points that have never been included before.

https://docs.rs/easy-ml/1.8.1/easy_ml/index.html
https://docs.rs/easy-ml/1.8.0/easy_ml/index.html

When I generate the docs locally on the 1.8.1 source code using the latest stable version of rust I see the same formatting for the 1.8.0 that docs.rs generated, so I think this is a bug that's slipped into the nightly version of docs.rs?

Stable local docs:
![Screenshot 2022-01-25 at 22-08-57 easy_ml - Rust](https://user-images.githubusercontent.com/13869843/151068667-ca986456-2da7-4398-b879-910878ea530b.png)
docs.rs:
![Screenshot 2022-01-25 at 22-08-43 easy_ml - Rust](https://user-images.githubusercontent.com/13869843/151068668-32d7e30d-6f46-417d-9276-a5196ea96dfe.png)

My guess is that it's because I'm using block comments, the unwanted bullet points seem to match up with empty lines in the documentation comment which consist of just a `*`.

```rust
/*!
 * If this is your first time using Easy ML you should check out some of the examples
 * to get an overview of how to use matrices then study the
 * [Matrix](matrices::Matrix) type for what you need.
 *
 * # Examples
 * - [Linear Regression](linear_regression)
 * - [k-means Clustering](k_means)
 * - [Logistic Regression](logistic_regression)
 * - [Naïve Bayes](naive_bayes)
 * - [Neural Network XOR Problem](neural_networks)
 *
 * # API Modules
 * - [Matrices](matrices)
 * - [Linear Algebra](linear_algebra)
 * - [Distributions](distributions)
 * - [(Automatic) Differentiation](differentiation)
 * - [Numerical type definitions](numeric)
 *
 * # Miscellaneous
 * - [Web Assembly](web_assembly)
 * - [SARSA and Q-learning using a Matrix for a grid world](sarsa)
 * - [Using custom numeric types](using_custom_types)
 */
```

Thanks

**Comment by Nemo157** [comment]
Seems potentially related to <https://github.com/rust-lang/rust/issues/92872>, but the latest builds were from a nightly that should have that fixed in; cc @GuillaumeGomez you probably have some idea on where the issue is.

**Comment by GuillaumeGomez** [comment]
This is highly surprising indeed.

@Skeletonxf Can you run `cargo doc` with the last nightly locally and tell us if it reproduces the bug please? If so, I'll try to check what's wrong in rustdoc.

**Comment by Skeletonxf** [comment]
@GuillaumeGomez 
I have ran cargo doc on the latest nightly locally (nightly-x86_64-unknown-linux-gnu rustc 1.60.0-nightly (777bb86bc 2022-01-20)) and it has reproduced the bug - I see the same generated docs as from the docs.rs build.

**Comment by GuillaumeGomez** [comment]
It'll be fixed by #93309.

**Comment by GuillaumeGomez** [comment]
Sure. But that for that, please open an issue on [docs.rs](github.com/rust-lang/docs.rs) repository directly.
